examples/gio_dbus_register_object/main.rs

```diff
@@ -8,10 +11,48 @@ const EXAMPLE_XML: &str = r#"
         <arg type='s' name='name' direction='in'/>
         <arg type='s' name='greet' direction='out'/>
       </method>
+      <method name='SlowHello'>
+        <arg type='s' name='name' direction='in'/>
+        <arg type='u' name='delay' direction='in'/>
+        <arg type='s' name='greet' direction='out'/>
+      </method>
     </interface>
   </node>
 "#;
 
+#[derive(Debug, glib::Variant)]
+struct Hello {
+    name: String,
+}
+
+#[derive(Debug, glib::Variant)]
+struct SlowHello {
+    name: String,
+    delay: u32,
+}
+
+#[derive(Debug)]
+enum HelloMethod {
+    Hello(Hello),
+    SlowHello(SlowHello),
+}
+
+impl DBusMethodCall for HelloMethod {
```

Comment:
In theory, this trait could get a derive macro, to automatically create the `match` from the names of the enum variants.

But I must admit that I'm not at all familiar with derive macros in Rust; I've never written one, and I'm not sure I'm going to find the time to learn that in near future.

So you folks consider this pull request a valuable API already, I'd appreciate if it was merged without a derive macro, as I'd like to leave that for a much later time, or perhaps even someone else :innocent: 

Hence,